tests/compile-fail/dlist.rs

```diff
@@ -10,5 +10,5 @@ pub fn test(_: LinkedList<u8>) {  //~ ERROR I see you're using a LinkedList!
 }
 
 fn main(){
-    test(LinkedList::new());
+    test(LinkedList::new()); //~ ERROR I see you're using a LinkedList!
```

Comment:
I thought this wasn't here on purpose: if `test` takes a `LinkedList`, the user has no choice but to build one to call the function. Hence the function should be warned about, not its call.